### ISSUE TITLE
Capability: Allow default and custom map furniture

### ISSUE THREAD

**Issue by prushforth** [Maps4HTML/HTML-Map-Element-UseCases-Requirements#16]
Using the `<video>` element as an example, the browser renderer is capable of rendering video content on demand of the author using something as simple as 

```
<video controls width="250">

    <source src="/media/examples/flower.webm"
            type="video/webm">

    <source src="/media/examples/flower.mp4"
            type="video/mp4">

    Sorry, your browser doesn't support embedded videos.
</video>
```

The video player rendered by that has default controls, styled according to the particular browser implementation.  Yet, and I am not an expert, but I believe you can shut the controls off (by omitting the `controls` attribute) and provide your own controls (if you wish) which respond to events (handled by script).

Maps should follow a similar approach: if you don't want default controls, don't ask for them.  Then, use events emitted by the player to bind your own self-created controls to the map, with script.

**Comment by Malvoz** [comment]
> I believe you can shut the controls off (by omitting the controls attribute) and provide your own controls (if you wish)

Correct, https://html.spec.whatwg.org/multipage/media.html#attr-media-controls

In addition to `controls`, consider [`controlslist`](https://wicg.github.io/controls-list/html-output/multipage/embedded-content.html#attr-media-controlslist) aswell.

**Comment by prushforth** [comment]
> In addition to controls, consider controlslist aswell.

Thanks I was not aware of that attribute. There will be a map-specific list of controls that we will need to define and specify.

**Comment by AmeliaBR** [comment]
The `controlslist` proposal is also new to me, but definitely relevant to the discussion (& should be mentioned in the related web standards subsection). There's a [PR under discussion for WHATWG HTML](https://github.com/whatwg/html/pull/2426), and a [matching issue](https://github.com/whatwg/html/issues/2293), in addition to the [WICG repository](https://wicg.github.io/controls-list/).

That said, the way it is currently defined (as a block list of `noXXX` keywords) is probably more narrow than what we want here.

**Comment by Malvoz** [comment]
@AmeliaBR 

> the way it is currently defined (as a block list of `noXXX` keywords) is probably more narrow than what we want here.

Help me understand why that is too narrow. `controls` enables default controls, `controlslist="noXXX"` disables specific controls that aren't desired. 

Is it too narrow because the presence of certain controls is a requirement? (e.g. similar to how e.g. [Chrome UA stylesheet](https://chromium.googlesource.com/chromium/blink/+/master/Source/core/css/mediaControls.css#30) does `audio:not([controls]) {display:none}` per WHATWG `<audio>` [rendering rules](https://html.spec.whatwg.org/multipage/rendering.html#embedded-content-rendering-rules)). I'd suppose a [static map](https://github.com/Maps4HTML/HTML-Map-Element-UseCases-Requirements/issues/17) would have no such requirements (which seems to be suggested in that issue). So I assume it is not this.

Or because for maps there is a desire to have a set of primary controls as "default" for when authors specify `controls`, and then somehow declaratively add "secondary" controls as not to "clutter" the map view with all available controls? To avoid such cluttering, browsers tend to have a context menu for controls such as a "download" button. I would also have to assume this is not the issue either.

**Comment by AmeliaBR** [comment]
> Help me understand why that is too narrow.

The problem comes when you want to have both default and custom controls as part of the map widget layout (as discussed in #79 ). HTML video doesn't currently have this option: you can create buttons on your web page that affect the video, but you can't add buttons to the video control panel itself.

To match how the libraries currently do it, you'd need to be able to specify which default controls get shown in which corners of the map.
